parsers/AM.py

```diff
@@ -0,0 +1,178 @@
+    gas_total = float(gas_tes[0])
+
+    hydro_ges = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[11])
+    hydro_altern = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[8])
+    hydro_total = float(hydro_ges[0])+float(hydro_altern[0])
+
+    nuclear_atom = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[9])
+    nuclear_total = float(nuclear_atom[0])
+
+    time_string = dparser.parse(data_split[14].split()[3],fuzzy=True)
+    date_time = time_string.replace(tzinfo=tz.gettz('Asia/Yerevan'))
+
+    #Operating solar, wind and biomass plants exist in small numbers, but are not reported yet
+    data = {
+        'zoneKey': zone_key,
+        'datetime': date_time,
+        'production': {
+            'gas': gas_total,
+            'hydro': hydro_total,
+            'nuclear': nuclear_total,
+            'coal': 0,
+            'geothermal': 0,
+            'oil': 0
+            },
+        'storage': {
+            'hydro storage': 0,
+            'battery storage': 0
+            },
+        'source': 'http://epso.am/poweren.htm'
+    }
+
+    return data
+
+
+def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=None):
+    """Requests the last known power exchange (in MW) between two countries
+    Arguments:
+    zone_key (optional) -- used in case a parser is able to fetch multiple countries
+    session (optional)      -- request session passed in order to re-use an existing session
+    Return:
+    A dictionary in the form:
+    {
+      'sortedZoneKeys': 'DK->NO',
+      'datetime': '2017-01-01T00:00:00Z',
+      'netFlow': 0.0,
+      'source': 'mysource.com'
+    }
+    """
+
+    r = session or requests.session()
+    url = 'http://epso.am/poweren.htm'
+    response = r.get(url)
+    response.encoding = 'utf-8'
+    html_doc = response.text
+    soup = BeautifulSoup(html_doc, 'html.parser')
+
+    data_string = soup.find(text=re.compile('var'))
+    data_split = data_string.split('\r\n')
+    GE_1 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[1])
+    GE_2 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[2])
+    GE_3 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[3])
+    AZ_1 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[4])
+    AZ_2 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[5])
+    IR_1 = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[6])
+
+    AM_AZ = float(AZ_1[0])+float(AZ_2[0])
+    AM_GE = float(GE_1[0])+float(GE_2[0])+float(GE_3[0])
+    AM_IR = float(IR_1[0])
+
+    time_string = dparser.parse(data_split[14].split()[3],fuzzy=True)
+    date_time = time_string.replace(tzinfo=tz.gettz('Asia/Yerevan'))
+ 
+    if '->'.join(sorted([zone_key1, zone_key2])) == 'AM->AZ':
```

Comment:
you can do;

```python
sorted_keys = '->'.join(sorted([zone_key1, zone_key2]))
```
at the start of the function which lets you avoid all these if statements.